pkg/controller/core/localqueue_controller.go

```diff
@@ -105,34 +145,54 @@ func (r *LocalQueueReconciler) NotifyWorkloadUpdate(oldWl, newWl *kueue.Workload
 // +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=localqueues/finalizers,verbs=update
 
 func (r *LocalQueueReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	var queueObj kueue.LocalQueue
-	if err := r.client.Get(ctx, req.NamespacedName, &queueObj); err != nil {
+	var lq kueue.LocalQueue
+	if err := r.client.Get(ctx, req.NamespacedName, &lq); err != nil {
 		// we'll ignore not-found errors, since there is nothing to do.
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
 	log := ctrl.LoggerFrom(ctx)
 	log.V(2).Info("Reconcile LocalQueue")
 
-	if ptr.Deref(queueObj.Spec.StopPolicy, kueue.None) != kueue.None {
-		err := r.UpdateStatusIfChanged(ctx, &queueObj, metav1.ConditionFalse, StoppedReason, localQueueIsInactiveMsg)
+	if ptr.Deref(lq.Spec.StopPolicy, kueue.None) != kueue.None {
+		err := r.UpdateStatusIfChanged(ctx, &lq, metav1.ConditionFalse, StoppedReason, localQueueIsInactiveMsg)
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
 	var cq kueue.ClusterQueue
-	err := r.client.Get(ctx, client.ObjectKey{Name: string(queueObj.Spec.ClusterQueue)}, &cq)
-	if err != nil {
+	if err := r.client.Get(ctx, client.ObjectKey{Name: string(lq.Spec.ClusterQueue)}, &cq); err != nil {
 		if apierrors.IsNotFound(err) {
-			err = r.UpdateStatusIfChanged(ctx, &queueObj, metav1.ConditionFalse, "ClusterQueueDoesNotExist", clusterQueueIsInactiveMsg)
+			err = r.UpdateStatusIfChanged(ctx, &lq, metav1.ConditionFalse, "ClusterQueueDoesNotExist", clusterQueueIsInactiveMsg)
 		}
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 	if meta.IsStatusConditionTrue(cq.Status.Conditions, kueue.ClusterQueueActive) {
-		err = r.UpdateStatusIfChanged(ctx, &queueObj, metav1.ConditionTrue, "Ready", "Can submit new workloads to localQueue")
-		return ctrl.Result{}, client.IgnoreNotFound(err)
+		if err := r.UpdateStatusIfChanged(ctx, &lq, metav1.ConditionTrue, "Ready", "Can submit new workloads to localQueue"); err != nil {
+			return ctrl.Result{}, client.IgnoreNotFound(err)
+		}
+	} else {
+		if err := r.UpdateStatusIfChanged(ctx, &lq, metav1.ConditionFalse, clusterQueueIsInactiveReason, clusterQueueIsInactiveMsg); err != nil {
+			return ctrl.Result{}, client.IgnoreNotFound(err)
+		}
 	}
-	err = r.UpdateStatusIfChanged(ctx, &queueObj, metav1.ConditionFalse, clusterQueueIsInactiveReason, clusterQueueIsInactiveMsg)
-	return ctrl.Result{}, client.IgnoreNotFound(err)
+
+	if r.admissionFSConfig != nil {
+		if err := r.initializeAdmissionFsStatus(ctx, &lq); err != nil {
+			return ctrl.Result{}, client.IgnoreNotFound(err)
+		}
+		recheckAfter := r.clock.Now().Sub(lq.Status.FairSharingStatus.AdmissionFairSharingStatus.LastUpdate.Time)
+		if recheckAfter < r.admissionFSConfig.UsageSamplingInterval.Duration {
+			return ctrl.Result{RequeueAfter: recheckAfter}, nil
```

Comment:
Is this correct?
Suppose duration=5min, now is after 3min from the last, so 2min to the next.
Then, Now().Sub(Last) is 3min, so we should requeue after 2min?

Reply:
That's a good catch, it should be `interval-recheckAfter`, I'll fix

Reply:
Fixed

Reply:
is there a test added to catch that bug? I would like to add such a test case, but it can be a follow up